health-services/libraries/health-services-common/src/main/java/org/egov/common/data/query/builder/GenericQueryBuilder.java

```diff
@@ -68,6 +69,20 @@ static List<String> getFieldsWithCondition(Object object, QueryFieldChecker chec
                             String fieldName = field.getName();
                             paramsMap.put(fieldName, field.get(object));
                             whereClauses.add(String.format("%s=:%s", fieldName, fieldName));
+                        } else if (field.getType().isAssignableFrom(ArrayList.class)
+                                && field.getGenericType() instanceof ParameterizedType
+                                && ((ParameterizedType) field.getGenericType()).getActualTypeArguments()[0].equals(String.class)) {
+                            ArrayList<String> arrayList = (ArrayList<String>) field.get(object);
+                            if (arrayList != null && !arrayList.isEmpty()) {
+                                String fieldName = field.getName();
+                                StringBuilder value = new StringBuilder();
+                                for (String arrayValue : arrayList) {
+                                    String paramName = fieldName + "_" + arrayList.indexOf(arrayValue);
```

Comment:
what happens if a value in the list is repeated by the caller? Might be safer to use old style for loop here rather than doing an `indexOf` inside the loop

Reply:
Updated

Reply:
Fix looks good.. could also use a ternary to perform the appends with or without comma but that is more for code readability.

Could you also please add a test case for a list with repeated values as well so any future refactoring/logic changes do not break this?